go/ct/gen/code.go

```diff
@@ -143,6 +143,11 @@ func (g *CodeGenerator) Generate(assignment Assignment, rnd *rand.Rand) (*st.Cod
 		// extend the runtime but are expected to reveal limited extra code coverage.
 		const expectedSize float64 = 200
 		size = int(rnd.ExpFloat64()/(1/expectedSize)) + minSize
+		// if there are data constraints, we need at least 2 bytes,
+		// one for an op, and the other for data.
+		if len(g.varIsDataConstraints) > 0 && size < 2 {
```

Comment:
I would like to see a unit test in `code_test.go` that fails before this modification and passes after.